crates/gitbutler-core/tests/suite/virtual_branches/create_commit.rs

```diff
@@ -221,43 +182,167 @@ async fn should_double_lock() {
+    request_opts.selected_for_changes = Some(true);
+    controller
+        .create_virtual_branch(project_id, &request_opts)
+        .await
+        .unwrap();
+
+    lines[6] = "change 2".to_string();
+    write_file(repository, "file.txt", &lines);
+
+    let commit_2 = controller
+        .create_commit(project_id, &branch_1_id, "commit 2", None, false)
+        .await
+        .unwrap();
+
+    lines[3] = "change3".to_string();
+    write_file(repository, "file.txt", &lines);
+
+    let branch = get_virtual_branch(controller, project_id, branch_1_id).await;
+    let locks = &branch.files[0].hunks[0].locked_to.clone().unwrap();
+
+    assert_eq!(locks.len(), 2);
+    assert_eq!(locks[0], commit_1);
+    assert_eq!(locks[1], commit_2);
+}
+
+#[tokio::test]
+async fn should_reset_into_same_branch() {
+    let Test {
+        project_id,
+        controller,
+        repository,
+        ..
+    } = &Test::default();
+
+    let mut lines = gen_file(repository, "file.txt", 7);
+    commit_and_push_initial(repository);
+
+    let base_branch = controller
+        .set_base_branch(project_id, &"refs/remotes/origin/master".parse().unwrap())
+        .await
+        .unwrap();
+
+    let branch_1_id = controller
+        .create_virtual_branch(project_id, &branch::BranchCreateRequest::default())
+        .await
+        .unwrap();
+
+    let mut request_opts = branch::BranchCreateRequest::default();
+    request_opts.selected_for_changes = Some(true);
+    let branch_2_id = controller
+        .create_virtual_branch(project_id, &request_opts)
+        .await
+        .unwrap();
+
+    lines[0] = "change 1".to_string();
+    write_file(repository, "file.txt", &lines);
+
+    controller
+        .create_commit(project_id, &branch_1_id, "commit 1", None, false)
+        .await
+        .unwrap();
+
+    controller
+        .reset_virtual_branch(project_id, &branch_2_id, base_branch.base_sha)
+        .await
+        .unwrap();
+
+    let files = get_virtual_branch(controller, project_id, branch_2_id)
+        .await
+        .files;
+    assert_eq!(files.len(), 1);
+}
+
+// TODO: Where should these helper functions live?
+
+fn write_file(repository: &TestProject, path: &str, lines: &Vec<String>) {
+    fs::write(repository.path().join(path), lines.join("\n")).unwrap()
+}
+
+fn gen_file(repository: &TestProject, path: &str, line_count: i32) -> Vec<String> {
+    let lines: Vec<_> = (0_i32..line_count).map(|i| format!("line {}", i)).collect();
+    write_file(repository, path, &lines);
+    lines
+}
+
+fn commit_and_push_initial(repository: &TestProject) {
+    repository.commit_all("initial commit");
+    repository.push();
+}
+
+async fn get_virtual_branch(
+    controller: &Controller,
+    project_id: &ProjectId,
+    branch_id: Id<Branch>,
+) -> VirtualBranch {
+    controller
+        .list_virtual_branches(project_id)
+        .await
+        .unwrap()
+        .0
+        .into_iter()
+        .find(|b| b.id == branch_id)
+        .unwrap()
```

Comment:
@krlvi I see you have a `get_branch` function in `VirtualBranchesHandle`. Could we make tests use this struct as well?